scripts/start.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script starts up a development server running Oppia. It installs any
+missing third-party dependencies and starts up a local GAE development
+server.
+"""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import argparse
+import atexit
+import fileinput
+import os
+import re
+import subprocess
+import sys
+import time
+
+import python_utils
+
+from . import build
+from . import common
+from . import install_third_party_libs
+from . import setup
+from . import setup_gae
+
+_PARSER = argparse.ArgumentParser(description="""
+    Run the script from the oppia root folder:
+        python -m scripts.start
+
+    Note that the root folder MUST be named 'oppia'.""")
+
+_PARSER.add_argument(
+    '--save_datastore',
+    help='optional; if specified, does not clear the datastore.',
+    action='store_true')
+_PARSER.add_argument(
+    '--enable_console',
+    help='optional; if specified, enables console.',
+    action='store_true')
+_PARSER.add_argument(
+    '--prod_env',
+    help='optional; if specified, runs Oppia in a production environment.',
+    action='store_true')
+_PARSER.add_argument(
+    '--no_browser',
+    help='optional; if specified, does not open a browser.',
+    action='store_true')
+
+PORT_NUMBER_FOR_GAE_SERVER = 8181
+
+
+def cleanup():
+    """Function for waiting for the servers to go down."""
+    common.print_each_string_after_two_new_lines([
+        'INFORMATION',
+        'Cleaning up the servers.'])
+    while common.is_port_open(PORT_NUMBER_FOR_GAE_SERVER):
+        time.sleep(1)
+
+
+def main(argv=None):
+    """Starts up a development server running Oppia."""
+    setup.main()
+    setup_gae.main()
```

Comment:
Explicitly pass empty args

Reply:
Setup and setup_gae accepts no arguments and hence its main accepts no argv

Reply:
That's why you should be passing an empty argument list. This is currently accepting `sys.argv` which will be an array with the filename at the front.

Reply:
I meant setup.py  and setup_gae.py does not accept any arguments. There is no argparse and parse_args in setup.py. Its never run with any arguments.

Reply:
I see, in that case add argparse into them anyway for consistency. You can add the script instructions/description into it so that it's not totally useless (it'll be nice to have `-h` for printing help).

Reply:
Hmm then we will be getting pylint errors that the variable _PARSER is never used. I think we can add them later if any flags are introduced in the setup file

Reply:
Why would it be unused? Just call `parse_args` like you would any other parser, and ignore the return value.

Reply:
Done